lib/spack/llnl/util/cpu/microarchitecture.py

```diff
@@ -191,23 +191,12 @@ def to_dict(self, return_list_of_items=False):
 
         return dict(list_of_items)
 
-    def optimization_flags(self, compiler, version):
-        """Returns a string containing the optimization flags that needs
-        to be used to produce code optimized for this micro-architecture.
-
-        If there is no information on the compiler passed as argument the
-        function returns an empty string. If it is known that the compiler
-        version we want to use does not support this architecture the function
-        raises an exception.
-
-        Args:
-            compiler (str): name of the compiler to be used
-            version (str): version of the compiler to be used
-        """
-        # If we don't have information on compiler at all
-        # return an empty string
+    def compiler_entry(self, compiler, version):
```

Comment:
`compiler_entry` is broken out so that Rust packages can extract the microarchitecture names for "clang" since the arch names in clang are applicable to Rust, too.